### PR TITLE
Add documentation for fair sharing

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Make a concept page just for preemption and explain both classic preemption and fair sharing.

#### Which issue(s) this PR fixes:

Part of #1714

#### Special notes for your reviewer:

Preview of new page https://deploy-preview-2322--kubernetes-sigs-kueue.netlify.app/docs/concepts/preemption/#fair-sharing

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/concepts/preemption.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+In a preemption, preempted Workloads are also called preemptees, and the ClusterQueue that
+the Workload belongs to is called a target ClusterQueue.
+Converselly, the Workload being accommodated is called the preemptor, and the ClusterQueue that the
+Workload belongs to is called a preempting ClusterQueue.
+
+## Reasons for preemption
+
+A Workload can preempt one or more Workloads if it is admitted in a [ClusterQueue with preemption enabled](/docs/concepts/cluster_queue/#preemption)
+and:
```
Comment (cortespao) [inline site/content/en/docs/concepts/preemption.md:17]: Do all the items in the list have to be true for the condition to be met? Or is it any?

I think that you should clarify by saying:

"(...) and any of the following events happen" OR "(...) and all the following events happen"

File: site/content/en/docs/concepts/preemption.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+### Candidates
+
+The list of preemption candidates is compiled from Workloads within the Cluster
+Queue satisfying the `withinClusterQueue` policy, and Workloads within the
+cohort which satisfy the `reclaimWithinCohort` policy.
+
+The list of candidates is sorted based on the following preference checks for
+tie-breaking:
+- Workloads from borrowing queues in the cohort,
+- Workloads with the lowest priority,
```
Comment (cortespao) [inline site/content/en/docs/concepts/preemption.md:71]: I'd remove the commas from these items.

File: site/content/en/docs/concepts/preemption.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    type: Evicted
+```
+
+## Preemption algorithms
+
+Kueue offers two preemption algorithms. The main difference between them is the criteria to allow
+preemptions from a ClusterQueue to others in the Cohort, when the usage of the preempting ClusterQueue is
+already above the nominal quota. The algorithms are:
+
+- **[Classic](#classic-preemption)**: Preemption in the cohort only happens when the usage of the preempting ClusterQueue
```
Comment (cortespao) [inline site/content/en/docs/concepts/preemption.md:45]:
```suggestion
- **[Classic Preemption](#classic-preemption)**: Preemption in the cohort only happens when the usage of the preempting ClusterQueue
```

File: site/content/en/docs/concepts/preemption.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+The list of candidates is sorted based on the following preference checks for
+tie-breaking:
+- Workloads from borrowing queues in the cohort,
+- Workloads with the lowest priority,
+- Workloads which got admitted the most recently.
+
+### Targets
+
+The algorithm qualifies the candidates as preemption targets using the heuristics
```
Comment (cortespao) [inline site/content/en/docs/concepts/preemption.md:76]:
```suggestion
The Classic Preemption algorithm qualifies the candidates as preemption targets using the heuristics
```

File: site/content/en/docs/concepts/preemption.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+3. If the current usage of the target queue is below nominal quota, then
+Kueue greedily qualifies the candidates, until the preemptor Workload can fit,
+disallowing for borrowing.
+
+4. If the Workload didn't fit by using the previous heuristics, Kueue greedily
+qualifies only the candidates which belong to the preempting Cluster Queue,
+until the preemptor Workload can fit, allowing for borrowing.
+
+The last step of the algorithm is to minimize the set of targets. For this
```
Comment (cortespao) [inline site/content/en/docs/concepts/preemption.md:96]:
```suggestion
5. The last step of the algorithm is to minimize the set of targets. For this
```

File: site/content/en/docs/concepts/preemption.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+To enable fair sharing, [use a Kueue Configuration](/docs/installation#install-a-custom-configured-release-version) similar to the following:
+
+```yaml
+apiVersion: config.kueue.x-k8s.io/v1beta1
+kind: Configuration
+fairSharing:
+  enable: true
+  preemptionStrategies: [LessThanOrEqualToFinalShare, LessThanInitialShare]
+```
+
```
Comment (cortespao) [inline site/content/en/docs/concepts/preemption.md:116]: The attributes in this Kueue configuration are described in the following sections.

**Review by cortespao — COMMENTED** [review]
Sending some suggestions. Overall LGTM

File: site/content/en/docs/concepts/cluster_queue.md [hunk 7]
```diff
@@ -56,6 +56,13 @@ For each resource, you can define quotas for multiple _flavors_.
 Flavors represent different variations of a resource (for example, different GPU
 models). You can define a flavor using a [ResourceFlavor object](/docs/concepts/resource_flavor).
 
+When definining quotas for a ClusterQueue, you can set the following values:
+- `nominalQuota` is the quantity of this resource that is available to a ClusterQueue at a point in time.
+- `borrowingLimit` is the maximum amount of quota that this ClusterQueue is allowed to borrow from the unused
+	nominal quota of other ClusterQueues in the same [cohort](#cohort).
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/cluster_queue.md:62]:
```suggestion
   nominal quota of other ClusterQueues in the same [cohort](#cohort).
```

File: site/content/en/docs/concepts/cluster_queue.md [hunk 8]
```diff
@@ -453,54 +460,8 @@ The fields above do the following:
...
-disallowing for borrowing.
-
-4. Kueue tries to greedily qualifies a subset of candidates which belong to the
-target Cluster Queue, until the incoming Workload can fit, allowing for borrowing.
-
-The last step of the algorithm is to optimize the set of targets. For this
-purpose Kueue greedily traverses the list of initial targets in reverse and
-removes them from the list of targets if the incoming Workload still can be
-admitted when they are accounted back for quota usage.
+Read [Preemption](/docs/concepts/preemption) to lear more about
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/cluster_queue.md:463]:
```suggestion
Read [Preemption](/docs/concepts/preemption) to learn more about
```

File: site/content/en/docs/concepts/preemption.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+```yaml
+status:
+  conditions:
+  - lastTransitionTime: "2024-05-29T14:37:19Z"
+    message: Preempted to accommodate a higher priority Workload
+    reason: Preempted
+    status: "True"
+    type: Evicted
+```
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/preemption.md:37]: Could we also add the type=Preempted condition?

File: site/content/en/docs/concepts/preemption.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+### ClusterQueue share value
+
+When you enable fair sharing, Kueue assigns a numeric share value to each ClusterQueue to summarize
+the usage of borrowed resources in a ClusterQueue, in comparisson to others in the same cohort.
+The share value is weighted by the `.spec.fairSharing.weight` defined in a ClusterQueue.
+
+During admission, Kueue prefers to admit Workloads from ClusterQueues that have the lowest share value first.
+During preemption, Kueue prefers to preempt Workloads from ClusterQueues that have the highest share value first.
+
+You can obtain the share value of a ClusterQueue in the `.spec.fairSharing.weightedShare` field or querying
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/preemption.md:126]:
```suggestion
You can obtain the share value of a ClusterQueue in the `.status.fairSharing.weightedShare` field or querying
```

😉

**Review by tenzen-y — COMMENTED** [review]
Generally, lgtm

File: site/content/en/docs/concepts/_index.md [hunk 11]
```diff
@@ -69,3 +69,9 @@ A group of ClusterQueues that can borrow unused quota from each other.
 The time between a Workload is created until it is admitted by a ClusterQueue.
 Typically, the Workload will compete with other Workloads for available
 quota based on the fair sharing rules of the ClusterQueue.
+
+### [Preemption](/docs/concepts/preemption)
+
+The process of evicting one or more admitted Workloads to accommodate another Workload.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply site/content/en/docs/concepts/_index.md]: Done

File: site/content/en/docs/concepts/cluster_queue.md [hunk 12]
```diff
@@ -56,6 +56,13 @@ For each resource, you can define quotas for multiple _flavors_.
 Flavors represent different variations of a resource (for example, different GPU
 models). You can define a flavor using a [ResourceFlavor object](/docs/concepts/resource_flavor).
 
+When definining quotas for a ClusterQueue, you can set the following values:
+- `nominalQuota` is the quantity of this resource that is available to a ClusterQueue at a point in time.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md]: Done

File: site/content/en/docs/concepts/cluster_queue.md (hunk 7 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md]: Oops, fixed

File: site/content/en/docs/concepts/cluster_queue.md [hunk 13]
```diff
@@ -56,6 +56,13 @@ For each resource, you can define quotas for multiple _flavors_.
...
+- `borrowingLimit` is the maximum amount of quota that this ClusterQueue is allowed to borrow from the unused
+	nominal quota of other ClusterQueues in the same [cohort](#cohort).
+- `lendingLimit` is the maximum amount of quota that this ClusterQueue allows other
+  ClusterQueues in the cohort to borrow when this ClusterQueue is not using it.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md]: I used "its nominal quota"

File: site/content/en/docs/concepts/preemption.md [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+date: 2024-05-28
+weight: 7
+description: >
+  Preemption is the process of evicting one or more admitted Workloads to accommodate another Workload.
+---
+
+In a preemption, preempted Workloads are also called preemptees, and the ClusterQueue that
+the Workload belongs to is called a target ClusterQueue.
+Converselly, the Workload being accommodated is called the preemptor, and the ClusterQueue that the
+Workload belongs to is called a preempting ClusterQueue.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md]: Oh, so much better. I still have plenty to learn :)

File: site/content/en/docs/concepts/preemption.md (hunk 1 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md]: It is any, thanks!

File: site/content/en/docs/concepts/preemption.md (hunk 9 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md:46]: Good idea. I rerun using kueue v0.7 and updated the status.

File: site/content/en/docs/concepts/preemption.md (hunk 3 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md]: Done

File: site/content/en/docs/concepts/preemption.md (hunk 2 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md]: Done

File: site/content/en/docs/concepts/preemption.md (hunk 4 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md]: Done

File: site/content/en/docs/concepts/preemption.md (hunk 5 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md:108]: 1 to 4 are not steps one after each other. They are different possible paths that the algorithm can take.

File: site/content/en/docs/concepts/preemption.md (hunk 10 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md]: Thanks!

File: site/content/en/docs/concepts/preemption.md (hunk 6 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/preemption.md:128]: Added

**Review by tenzen-y — COMMENTED** [review]
Thanks!
/lgtm

**Pull request merged**